xiaomi.vacuum.gen1/firmwarebuilder/flasher.py

```diff
@@ -1,12 +1,41 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
-import re
-import os
-import subprocess
-import time
 import socket
 import hashlib
 import json
+import sys
+import argparse
+import http.server
+import socketserver
+import threading
+from time import sleep
+import miio
+
+
+class ThreadedHTTPServer(socketserver.ThreadingMixIn, http.server.HTTPServer):
+    daemon_threads = True
+
+
+def printProgressBar (iteration, total, prefix = '', suffix = '', decimals = 1, length = 100, fill = '█'):
+    """
+    Call in a loop to create terminal progress bar
+    @params:
+        iteration   - Required  : current iteration (Int)
+        total       - Required  : total iterations (Int)
+        prefix      - Optional  : prefix string (Str)
+        suffix      - Optional  : suffix string (Str)
+        decimals    - Optional  : positive number of decimals in percent complete (Int)
+        length      - Optional  : character length of bar (Int)
+        fill        - Optional  : bar fill character (Str)
+    """
+    percent = ("{0:." + str(decimals) + "f}").format(100 * (iteration / float(total)))
+    filledLength = int(length * iteration // total)
+    bar = fill * filledLength + '-' * (length - filledLength)
+    print('\r%s |%s| %s%% %s' % (prefix, bar, percent, suffix), end = '\r')
+    # Print New Line on Complete
+    if iteration == total:
+        print()
```

Comment:
not a biggie, but you may be interest in tqdm or progressbar libs instead of implementing progress bars on your own in the future.

Reply:
Thanks, I'll check it out

Reply:
Should be added: code was taken from: https://stackoverflow.com/questions/3173320/text-progress-bar-in-the-console/34325723#34325723

But still, less dependencies is better, or not?

Reply:
Well, that's up to the project maintainer, differenet people have different preferencies. It's not a biggie anyway, just wanted to let you know that there exist nice implementations in pypi for such :-)